### PR TITLE
Implemented shares/uncles position chart on status command

### PR DESCRIPTION
**Pull request by WeebDataHoarder**
Shows progress in slices of PPNLS window. Shares move left to right as they age, right being the end of the PPNLS window.

A dot `.` indicates no shares, `1-9` for that number in shares, for 9+ it shows a `+`.
Will not show shares or uncles lines unless you have one on window.

```
Main chain height         = 2446385
Main chain hashrate       = 2.645 GH/s
Side chain height         = 87802
Side chain hashrate       = 52.269 MH/s
Your hashrate (pool-side) = 167.221 KH/s
PPLNS window              = 2160 blocks (+70 uncles, 0 orphans)
Your shares               = 6 blocks (+1 uncles, 0 orphans)
Your shares in window     = [1.....2...1...1...1...........]
Your uncles in window     = [........1.....................]
Block reward share        = 0.320% (0.002789310611 XMR)
```

### CODE REVIEW

File: src/side_chain.cpp [hunk 1]
```diff
@@ -598,15 +598,18 @@ void SideChain::print_status()
...
 
 	while (cur) {
 		blocks_in_window.emplace_back(cur->m_sidechainId);
 		++total_blocks_in_window;
 
 		if (cur->m_minerWallet == m_pool->params().m_wallet) {
-			++our_blocks_in_window;
+			// this produces an integer division with quotient rounded up, avoids non-whole divisions from overflowing on total_blocks_in_window
+			const size_t window_index = (total_blocks_in_window - 1) / ((m_chainWindowSize + our_blocks_in_window.size() - 1) / our_blocks_in_window.size());
+			our_blocks_in_window[std::min(window_index, our_uncles_in_window.size() - 1)]++; // clamp window_index, even if total_blocks_in_window is not larger than m_chainWindowSize
```
Comment (SChernykh) [inline src/side_chain.cpp:612]: our_uncles_in_window.size() - copy/paste error?

File: src/side_chain.cpp [hunk 2]
```diff
@@ -598,15 +598,18 @@ void SideChain::print_status()
 	uint32_t total_blocks_in_window = 0;
 	uint32_t total_uncles_in_window = 0;
 
-	uint32_t our_blocks_in_window = 0;
-	uint32_t our_uncles_in_window = 0;
+	// each dot corresponds to m_chainWindowSize / 30 shares, with current values, 2160 / 30 = 72
+	std::vector<uint32_t> our_blocks_in_window(30, 0);
+	std::vector<uint32_t> our_uncles_in_window(30, 0);
```
Comment (SChernykh) [inline src/side_chain.cpp:603]: If it's fixed size, it's better to use std::array.

**Review by SChernykh — CHANGES_REQUESTED** [review]

File: src/side_chain.cpp (hunk 1 above)
Comment (WeebDataHoarder) [reply src/side_chain.cpp:612]: Correct, fixed a math error and copy&pasted, will force-push a change

File: src/side_chain.cpp (hunk 2 above)
Comment (WeebDataHoarder) [reply src/side_chain.cpp:603]: Will have them convert to `std::array<uint32_t, 30> our_blocks_in_window{};` for zero-initialization

**Review by SChernykh — APPROVED** [review]

**Pull request merged**